### ISSUE TITLE
Source Documents 

### ISSUE THREAD

**Issue by rogalvil** [ajndkr/fastapi-async-langchain#9]
I was testing with RetrievalQAWithSourcesChain and it is working fine but the document sources are not obtained, I wanted to ask you if you have thought about how to approach this issue.

Perhaps I should include them in the last message, but I don't know the best way to present that information.

**Comment by ajndkr** [comment]
Hi! do you maybe have an example of the output?

note that I have recently added a new response class for `RetrievalQA`. I haven't tested if it works with `RetrievalQAWithSourcesChain` yet. 

```
from fastapi_async_langchain.responses import RetrievalQAStreamingResponse
```

AFAIK, the sources are added in the end by the LLM so it should be accessible by the `on_llm_new_token` method of the callback class.

**Comment by rogalvil** [comment]
My output is this, I get sources on last part of message  but I would like get page content as is get when return_source_documents is true.

![Captura de pantalla 2023-04-25 a la(s) 13 21 10](https://user-images.githubusercontent.com/695923/234381580-a7d1bf99-21f0-49e3-8047-095d2b11f50a.png)

![Captura de pantalla 2023-04-25 a la(s) 13 20 15](https://user-images.githubusercontent.com/695923/234381795-4a718000-528a-4464-aa08-a633a64f47e6.png)

**Comment by ajndkr** [comment]
gotcha! one idea I have is to update the `on_chain_end` method of the `AsyncFastApiStreamingCallback` class and stream the source documents using the `send` attribute. Similar to what the `on_llm_new_token` method does.

Since this is an interesting feature, would you like to contribute?

**Comment by rogalvil** [comment]
Yes of course, let me try to implement this issue

**Comment by ajndkr** [comment]
@rogalvil your changes have been published in `0.3.2`. Please check if everything is working as expected. If all good, please close this issue. cheers!
